### PR TITLE
Added histogram plotting function

### PR DESCRIPTION
**Pull request by AditiThirdEye**
Implemented the task as asked in this issue #2722<br>
This function is implemented with the help of matplotlib lib as mentioned in the task.
Function to plot histogram 
 * User should provide at least the required parameters
 *
 * REQUIRED:
 * @param independent_var is the Input list which will provide the independent variable usually known as x.
 * @param dependent_var is the Input list which will provide the dependent variable usually known as y.
 *
 * EXTRA PARAMETERS: 
 * @param plot_lim_left the lowest number limit till the x-axis should be plotted on the graph.
 * @param plot_lim_right the highest number limit till the x-axis should be plotted on the graph.
 * @param size_len number of pixels in x-dimentions to be plotted in the output.
 * @param size_breath number of pixels in y-dimentions to be plotted in the output.
 * @param legend_name the name that should be given to the line in the plot.
 * @param plot_title the title that should be provided to the plot.
 * @param plot_name the name with the extension that the plot should be saved with.<br>
Please give me suggestion how can I improve my contribution.

Regards, 
Aditi

### CODE REVIEW

File: src/mlpack/core/data/histogram.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ */
+#ifndef MLPACK_CORE_DATA_HISTOGRAM_HPP
+#define MLPACK_CORE_DATA_HISTOGRAM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core.hpp>
+
+#include "matplotlibcpp.h"
+namespace plt = matplotlibcpp;
+
```
Comment (shrit) [inline src/mlpack/core/data/histogram.hpp:21]: @AditiThirdEye Thanks for your contribution, the idea is good, but there are several flows:
This code adds a direct dependency on an external library `matplotlib` inside mlpack core.
Matplotlibcpp requires the uses to have `matplotlib` python installed too, which will eventually make mlpack library dependent on python.
Our current objective is to remove the dependency on external library, unless it is very necessary with preference header-only.
I am not sure if this is the right place for this feature, nor what is going to add to mlpack library.
We have an example repository `mlpack/examples`, if you would like to show to plot data set using C++, I think `mlpack/examples` is the place to for it, even though the dependency seems to be quite large at the same time.

Comment (AditiThirdEye) [reply src/mlpack/core/data/histogram.hpp:21]: Hello @shrit , 
Actually in this #2722 
In the histogram task it was asked to make use of the matplotlibcpp lib
Hence I used it.

Comment (AditiThirdEye) [reply src/mlpack/core/data/histogram.hpp:21]: Also in the discussion of #2722 
You can find below that @heisenbuug  asked me to use matplotlib.cpp as a single header file
So can you please guide me how can I improve my contribution.

Regards,
Aditi

Comment (heisenbuug) [reply src/mlpack/core/data/histogram.hpp:21]: @shrit but if we plan on making something like a data frame we would need
to plot the data while preprocessing to get an insight into the data, like if it's skewed
or to look at outliers.

Should we use something else here? I know the idea is not actually totally clear here
and may require a to of discussion here.

Actually, I was looking into the visualizer project `mlboard` which we had last summer, maybe
we connect to that.

Comment (zoq) [reply src/mlpack/core/data/histogram.hpp:21]: I would also just put this into the examples repo, e.g. into the utils folder - https://github.com/mlpack/examples/tree/master/utils. I don't think there is need to add this to the core repo; users can always build mlpack and use utility functions from other sources. In this case if I like to plot the histogram I could use the function from the examples repo, that has some other utility functions that are meant to visualize data.

Comment (AditiThirdEye) [reply src/mlpack/core/data/histogram.hpp:21]: Okay @zoq sure, I'll do that 👍🏼

Comment (heisenbuug) [reply src/mlpack/core/data/histogram.hpp:21]: Okay @zoq @shrit.
No need to add these to the core mlpack, that makes sense.

Thank you.

File: src/mlpack/core/data/histogram_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                    int size_len = 1200, int size_breath = 780,
+                    string legend_name = "Legend Name",
+                    string plot_title = "Plot Title";
+                    string plot_name = "./Plot_name.png";
+                    )
+ {
+
+
+    // Set the size of output image to 1200x780 pixels by default
+    plt::figure_size(size_len, size_breath);
```
Comment (heisenbuug) [inline src/mlpack/core/data/histogram_impl.hpp:54]: Thanks for the contribution.
We prefer different naming styles in mlpack.

You can go through [this](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions) document.

We use camel casing, so I believe `plot_title` should be `plotTitle`

**Pull request closed without merging**